CHANGELOG.md

```diff
@@ -1,3 +1,122 @@
+### citus v12.0.0 (July 11, 2023) ###
+
+PR #6866 : Add support for schema-based-sharding via a GUC
+TODO: PLEASE SHORTEN THE NEXT LINE MANUALLY, IT SHOULD BE NO LONGER THAN 78 CHARS
+* Adds citus.enable_schema_based_sharding GUC that allows sharding the database based on schemas when enabled.
+
+* Adds `citus_schema_distribute` and `citus_schema_undistribute` UDFs for
+  conversion between regular and distributed schemas (#6933)
+
+* Adds support for creating distributed tables without shard key (#6867)
+
+* Enable logical planner for single-shard tables (#6950)
+
+* Add citus_schemas view (#6979)
+
+* Enabling citus_stat_tenants to support schema-based tenants. (#6936)
+
+* Support CREATE TABLE .. AS SELECT .. commands for tenant tables (#6998)
+
+* Disallow single shard table creation if replication factor > 1 (#7006)
+
+PR #7027 : Improve error/hint messages related to schema-based sharding
+
+PR #6971 : Fix citus_table_type column in citus_tables and citus_shards views for single shard tables
+
+PR #6965 : Disable some udfs for tenant tables
+
+PR #6963 : Single Shard Misc UDFs 2
+
+PR #6924 : Add Support for Single Shard Tables in update_distributed_table_colocation
+
+PR #6916 : Citus UDFs support for single shard tables
+
+PR #6908 : Prevent downgrades when there is a single-shard table in the cluster
+
+PR #6902 : Disable master_create_empty_shard udf for single shard tables
+
+* Drops PG13 Support (#7002, #7007)
+
+PR #6975 : Citus Revise tree-walk APIs to include context
+
+PR #6974 : Random warning fixes
+
+PR #6957 : Fix some gucs' initial and boot values, and flag combinations
+
+* Support MERGE for non-colocated distributed tables and non-dist column joins. (#6927)
+
+* Fixes shard size bug with too many shards (#7018)
+
+* Improves citus_tables view performance (#7018)
+
+* Improve tenant monitoring performance (#6868)
+
+* Add CPU usage to citus_stat_tenants (#6844)
+
+* Optimize push-down planner on memory and cpu (#6945)
+
+PR #7024 : Improve error message for CREATE SCHEMA .. CREATE TABLE
+
+PR #7025 : Fix Reference Table Check for CDC
+
+PR #7026 : Add locking mechanism for tenant monitoring probabilistic approach
+
+* Change default rebalance strategy to by_disk_size (#7033)
+
+* Change by_disk_size rebalance strategy to have a base size (#7035)
+
+PR #6930 : Run replicate_reference_tables background task as superuser.
```

Comment:
@emelsimsek Could you shorten that descriptions for #6930, #6986, and #6983?

Reply:
What do you think about changing 6986' description like below?
"Fixes deparsing `ALTER TABLE .. ADD FOREIGN KEY` with schema qualified table"

And 6983 like below?
"The GUC `search_path` is now reported when it is changed"

Reply:
I liked your suggestions. But I tweaked the one for 6986 to make a bit more customer facing.